### PR TITLE
feat: update file feature_toggle_annotations.yaml

### PR DESCRIPTION
**Pull request by Yagnesh1998**
**Description:**
Add annotation for toggle_removal_ticket to make it simpler to know if a toggle has an associated DEPR or removal ticket.
As Per this DEPR: https://github.com/openedx/edx-toggles/issues/324
**Additional notes:**

This will be an optional annotation, at least to start.
Thoughts for documentation enhancements:
The link could be to a DEPR or another ticket.
Ideally it would not be a private ticket, but maybe that is ok if the removal is part of the same ticket used to introduce the toggle and it will be very short-lived?
In this case (or all cases?), all information that would support the removal should be added to the description or warning annotation.
We should note that for temporary rollout toggles, we should document the intended final value. Note that the toggle default might change as part of the rollout, which means it won't always just be the opposite.
This ticket was spun out of https://github.com/openedx/edx-toggles/issues/283, which listed a number of potential improvements.

### CODE REVIEW

**Review by robrap — COMMENTED** [review]
Great. Thanks. 
1. You also need to update the version to 1.6.0 and add a note to the CHANGELOG.
2. We may hold on merging until we get the how-to docs updated as well.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -14,6 +14,10 @@ Change Log
 Unreleased
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+[1.6.0] - 2024-01-31
```
Comment (robrap) [inline CHANGELOG.rst:17]: Note that this typically requires the version update in code as well. See https://github.com/openedx/code-annotations/blob/764c306d859036971177457cb08cc04e7ee2989c/code_annotations/__init__.py#L5. Thanks.

**Review by robrap — APPROVED** [review]
Thanks @Yagnesh1998.

**Comment by openedx-webhooks** [comment]
@Yagnesh1998 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/code-annotations&entry.867055334=https://github.com/openedx/code-annotations/pull/97&entry.1484655318=https://github.com/Yagnesh1998&entry.752974735=2024-01-17+12:57&entry.1917517419=2024-01-30+20:41&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**